src/geode/mesh/io/edged_curve_input.cpp

```diff
@@ -25,18 +25,48 @@
 
 #include <geode/mesh/core/edged_curve.h>
 
+namespace
+{
+    template < geode::index_t dimension >
+    void load( geode::EdgedCurve< dimension >& edged_curve,
+        absl::string_view filename )
+    {
+        auto input = geode::EdgedCurveInputFactory< dimension >::create(
+            geode::extension_from_filename( filename ).data(), edged_curve,
+            filename );
+        input->read();
+    }
+} // namespace
+
 namespace geode
 {
     template < index_t dimension >
-    void load_edged_curve(
-        EdgedCurve< dimension >& edged_curve, absl::string_view filename )
+    std::unique_ptr< EdgedCurve< dimension > > load_edged_curve(
+        const MeshType& type, absl::string_view filename )
     {
         try
         {
-            auto input = EdgedCurveInputFactory< dimension >::create(
-                extension_from_filename( filename ).data(), edged_curve,
-                filename );
-            input->read();
+            auto edged_curve = EdgedCurve< dimension >::create( type );
+            load( *edged_curve, filename );
+            return edged_curve;
+        }
+        catch( const OpenGeodeException& e )
+        {
+            Logger::error( e.what() );
+            throw OpenGeodeException{ "Cannot load EdgedCurve from file: ",
+                filename };
+        }
+    }
+
+    template < index_t dimension >
+    std::unique_ptr< EdgedCurve< dimension > > load_edged_curve(
+        absl::string_view filename )
+    {
+        try
+        {
+            auto edged_curve = EdgedCurve< dimension >::create();
+            load( *edged_curve, filename );
+            return edged_curve;
         }
         catch( const OpenGeodeException& e )
         {
```

Comment:
Should that kind of code be duplicated or `load_edged_curve` without mesh type should call the one with mesh type giving default one?

Reply:
In this case, you have to know in this file what the default is. By having the `create()`, this is this only location where the default behavior is defined.

Reply:
I do not understand. You can use OpenGeodeEdgedCurve< dimension >::type_name_static()?